pkg/listwatch/listwatch.go

```diff
@@ -36,31 +37,45 @@ import (
 // only v1.NamespaceAll, then this func assumes that the client has List and
 // Watch privileges and returns a regular cache.ListWatch, since there is no
 // other way to get all namespaces.
-func NewUnprivilegedNamespaceListWatchFromClient(c cache.Getter, namespaces []string, fieldSelector fields.Selector) *cache.ListWatch {
+//
+// The namespace whitelist and blacklist are mutually exclusive.
+// See NewFilteredUnprivilegedNamespaceListWatchFromClient for a description on how they are applied.
+func NewUnprivilegedNamespaceListWatchFromClient(c cache.Getter, nsWhitelist, nsBlacklist []string, fieldSelector fields.Selector, l log.Logger) cache.ListerWatcher {
 	optionsModifier := func(options *metav1.ListOptions) {
 		options.FieldSelector = fieldSelector.String()
 	}
-	return NewFilteredUnprivilegedNamespaceListWatchFromClient(c, namespaces, optionsModifier)
+	return NewFilteredUnprivilegedNamespaceListWatchFromClient(c, nsWhitelist, nsBlacklist, optionsModifier, l)
 }
 
 // NewFilteredUnprivilegedNamespaceListWatchFromClient mimics
 // cache.NewUnprivilegedNamespaceListWatchFromClient.
-// It allows for the creation of a cache.ListWatch for namespaces from a client
-// that does not have `List` privileges. If the slice of namespaces contains
-// only v1.NamespaceAll, then this func assumes that the client has List and
+// It allows for the creation of a cache.ListWatch for a whitelist or blacklist of namespaces
+// from a client that does not have `List` privileges.
+//
+// If the given namespace whitelist contains only v1.NamespaceAll,
+// then this function assumes that the client has List and
 // Watch privileges and returns a regular cache.ListWatch, since there is no
 // other way to get all namespaces.
-func NewFilteredUnprivilegedNamespaceListWatchFromClient(c cache.Getter, namespaces []string, optionsModifier func(options *metav1.ListOptions)) *cache.ListWatch {
+//
+// The namespace whitelist and blacklist are mutually exclusive.
+// If the namespace whitelist contains multiple items, the given blacklist has no effect.
+// If the namespace whitelist includes one entry with the value v1.NamespaceAll (empty string),
+// the given blacklist is being applied.
+func NewFilteredUnprivilegedNamespaceListWatchFromClient(c cache.Getter, nsWhitelist, nsBlacklist []string, optionsModifier func(options *metav1.ListOptions), l log.Logger) cache.ListerWatcher {
 	// If the only namespace given is `v1.NamespaceAll`, then this
```

Comment:
this comment should be adapted to reflect the new logic

Reply:
fixed